tests/features/S2N_LIBCRYPTO_SUPPORTS_ENGINE.c

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/*
+ * This feature probe checks if the linked libcrypto has ENGINE support [1].
+ *
+ * Normally, we would always expect the `openssl/engine.h` header to be
+ * available. However, some platforms (CentOS 10,  Fedora 41, and RHEL 10 are
+ * manually removing the `openssl/engine.h` header (for more detail look at
+ * issues [2] [3]).
+ *
+ * [1] https://docs.openssl.org/1.0.2/man3/engine/
+ * [2] https://github.com/aws/s2n-tls/pull/4705
+ * [3] https://github.com/aws/s2n-tls/pull/4873
+ */
+
+#include <openssl/engine.h>
+
+int main()
+{
+#if OPENSSL_NO_ENGINE
+    fail compilation : engine not supported
```

Comment:
you can emit an error with

```
#error "engine is not supported"
```

Reply:
I dont see that being emitted when I build s2n-tls. When does that actually happen?

Reply:
It doesn't get emitted by default, since probes are somewhat expected to fail.

Reply:
But you can enable it here: https://github.com/aws/s2n-tls/blob/12b140a98a60c7d43eb89f891eab8c7f8d3de614/CMakeLists.txt#L343